src/sonic-yang-models/yang-models/sonic-dtel.yang

```diff
@@ -0,0 +1,294 @@
+                    type uint16;
+                    description "DTEL flow state clear cycle, units: seconds";
+                }
+
+                leaf latency_sensitivity {
+                    type uint16;
+                    description "Latency sensitivity for flow state change detection, units: 2^n nanoseconds";
+                }
+
+                leaf-list sink_port_list {
+                    type union {
+                        type leafref {
+                            path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name";
+                        }
+                        type leafref {
+                            path "/lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name";
+                        }
+                    }
+                    default "";
+                    description "DTEL sink ports";
+                }
+
+                leaf int_endpoint {
+                    type string {
+                        pattern "true|false";
+                    }
+                    default "false";
+                }
+
+                leaf int_transit {
+                    type string {
+                        pattern "true|false";
+                    }
+                    default "false";
+                }
+                leaf queue_report {
+                    type string {
+                        pattern "true|false";
+                    }
+                    default "false";
+                }
+
+                leaf drop_report {
+                    type string {
+                        pattern "true|false";
+                    }
+                    default "false";
+                }
+
+                leaf postcard {
+                    type string {
+                        pattern "true|false";
+                    }
+                    default "false";
+                }
+
+            }
+        }
+
+        container DTEL_REPORT_SESSION {
+            description "DTEL report session part of config_db.json";
+
+            list DTEL_REPORT_SESSION_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..128;
+                    }
+                }
+
+                leaf src_ip {
+                    type inet:ip-address;
+                    description "DTEL report source IP address";
+                }
+
+                leaf-list dst_ip_list {
+                    type inet:ip-address;
+                    description "DTEL report destination IP addresses";
+                }
+
+                leaf truncate_size {
+                    type uint32;
+                    description "DTEL report truncate size";
+                }
+
+                leaf udp_dest_port {
+                    type string {
+                        pattern "0|[1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]";
+                    }
```

Comment:
Can you use type inet:port-number?

Reply:
Got it. Already fixed it.
Please review again.